File: app/src/main/java/se/gustavkarlsson/skylight/android/dagger/components/MainActivityComponent.java
package se.gustavkarlsson.skylight.android.dagger.components;

import dagger.Subcomponent;
import se.gustavkarlsson.skylight.android.dagger.modules.definitive.UpdateErrorBroadcastReceiverModule;
import se.gustavkarlsson.skylight.android.dagger.modules.replaceable.FragmentRootViewModule;
import se.gustavkarlsson.skylight.android.dagger.modules.replaceable.SwipeToRefreshModule;
import se.gustavkarlsson.skylight.android.dagger.scopes.ActivityScope;
import se.gustavkarlsson.skylight.android.gui.activities.main.MainActivity;

@SuppressWarnings("WeakerAccess")
@Subcomponent(modules = {
		SwipeToRefreshModule.class,
		UpdateErrorBroadcastReceiverModule.class
})
@ActivityScope
public interface MainActivityComponent {
	void inject(MainActivity mainActivity);

	AuroraChanceFragmentComponent getAuroraChanceFragmentComponent(FragmentRootViewModule fragmentRootViewModule);
	AuroraFactorsFragmentComponent getAuroraFactorsFragmentComponent(FragmentRootViewModule fragmentRootViewModule);
}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/dagger/components/MainActivityComponent.kt
package se.gustavkarlsson.skylight.android.dagger.components

import dagger.Subcomponent
import se.gustavkarlsson.skylight.android.dagger.modules.*
import se.gustavkarlsson.skylight.android.dagger.scopes.ActivityScope
import se.gustavkarlsson.skylight.android.gui.activities.main.MainActivity

@Subcomponent(modules = arrayOf(
	ActivityModule::class,
	SwipeToRefreshModule::class,
	PresentRecentAuroraReportModule::class,
	LastAuroraReportProviderModule::class,
	PresentingErrorsModule::class,
	SetUpdateScheduleModule::class,
	ForegroundReportLifetimeModule::class
))
@ActivityScope
interface MainActivityComponent {
	fun inject(mainActivity: MainActivity)

	fun getAuroraChanceFragmentComponent(fragmentRootViewModule: FragmentRootViewModule): AuroraChanceFragmentComponent
	fun getAuroraFactorsFragmentComponent(fragmentRootViewModule: FragmentRootViewModule): AuroraFactorsFragmentComponent
}
